src/main/java/com/exasol/releasedroid/adapter/repository/JavaRepositoryValidator.java

```diff
@@ -49,6 +50,17 @@ private Report validateVersion(final MavenPom mavenPom) {
         return report;
     }
 
+    private Report validateGroupId(final MavenPom mavenPom) {
+        final var report = ValidationReport.create();
+        if (mavenPom.hasGroupId()) {
+            report.addSuccessfulResult("'groupId' in the pom file exists.");
+        } else {
+            report.addFailedResult(ExaError.messageBuilder("E-RD-REP-31")
+                    .message("Cannot detect a 'groupId' in the pom file.").toString());
+        }
+        return report;
+    }
+
```

Comment:
I tjhink we had that a while ago here and tjhen removed it.
The idea was to remove all java-specific validations that can be done in project-keeper and only check that project-keeper is used.
The advantage of this approach is that problems are detected much earlier and not late then when you merged everything and want to release.

Reply:
On the other hand, you really need group id here. So maybe it's good to check if it's exsists.

Reply:
You are right, project keeper should check this.
But this validation allows use to have a better error handling than throwing a NullPointerException.